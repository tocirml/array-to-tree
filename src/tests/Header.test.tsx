import React from 'react';
import Header, { HeaderRenderer } from '../components/Header';
import { shallow } from 'enzyme';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

const TITLE = 'Tree Traversal';

describe('rendering', () => {
  it('renders without crashing', () => {
    render(<Header />);
  });

  it('renders correct title', () => {
    render(<Header />);
    expect(screen.getByTestId('header')).toHaveTextContent(TITLE);
  });
});

describe('snapshots', () => {
  it('Header snapshot', () => {
    const tree = shallow(<Header />);
    expect(tree).toMatchSnapshot();
  });
  it('HeaderRenderer snapshot', () => {
    const appState: any = {
      appState: {
        title: TITLE,
      },
    };
    const tree = shallow(<HeaderRenderer appState={appState} />);
    expect(tree).toMatchSnapshot();
  });
});
