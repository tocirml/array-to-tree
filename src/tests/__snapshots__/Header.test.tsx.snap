// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Header snapshot 1`] = `
<HeaderRenderer
  appState={
    AppState {
      "bodyMessage": "Process the input text to a BinTreeNode",
      "smallestTreeId": "root",
      "title": "Tree Traversal",
      "treeNode": BinTreeNode {
        "id": "root",
        "left": null,
        "right": null,
      },
    }
  }
/>
`;

exports[`snapshots HeaderRenderer snapshot 1`] = `
<header
  className="App-header"
  data-testid="header"
/>
`;
